scripts/o3de/o3de/manifest.py

```diff
@@ -221,9 +221,16 @@ def is_gem_subdirectory(subdir_files):
         return False
 
     gem_directories = []
+
     # Locate all subfolders with gem.json files within them
     if external_subdirs:
         for subdirectory in external_subdirs:
+            subdirectory_path = pathlib.PurePath(subdirectory).as_posix()
+
+            # don't revisit paths added when walking the directory tree
+            if subdirectory_path in gem_directories:
+                continue
```

Comment:
Actually this logic should be moved to the `get_*_external_subdirectories` functions as this is too high of a level to be dealing with duplicate directories.

Reply:
My suggestion is to remove this logic from `get_gems_from_external_subdirectories`.

We can then perform a separate pass on all the `get_*_projects`, `get_*_engines`, `get_*_templates`, `get_*_restricted`, `get_*_repos` and `get_*_external_subdirectories` methods within this file to use [dict.fromkeys](https://docs.python.org/3/library/stdtypes.html?highlight=fromkeys#dict.fromkeys) to filter out duplicate entries of those objects, while also maintaining the order they were within the list.

For example we would update `get_manifest_external_subdirectories` method to the following to filter out duplicates

```
def get_manifest_external_subdirectories() -> list:
    json_data = load_o3de_manifest()
return list(dict.fromkeys(json_data['external_subdirectories'])) if 'external_subdirectories' in json_data else []
```

Reply:
Actually what could do now that the gem.json supports the "external_subdirectories" key is to remove the walking of subdirectories and only visit gems directly specified in the "external_subdirectories" field of a gem.